src/core/Object3D.js

```diff
@@ -115,8 +216,95 @@ class Object3D extends EventDispatcher {
 
 		this.userData = {};
 
+		this.updateMatrixSideEffects();
+
+	}
+
+	set matrixAutoUpdate( value ) {
+
+		this.matrixData.matrixAutoUpdate = value;
+
+	}
+
+	get matrixAutoUpdate() {
+
+		return this.matrixData.matrixAutoUpdate;
+
+	}
+
+	set matrixWorldNeedsUpdate( value ) {
+
+		this.matrixData.matrixWorldNeedsUpdate = value;
+
+	}
+
+	get matrixWorldNeedsUpdate() {
+
+		return this.matrixData.matrixWorldNeedsUpdate;
+
+	}
+
+	set matrixWorldAutoUpdate( value ) {
+
+		this.matrixData.matrixWorldAutoUpdate = value;
+
+	}
+
+	get matrixWorldAutoUpdate() {
+
+		return this.matrixData.matrixWorldAutoUpdate;
+
+	}
+
+	updateMatrixSideEffects() {
+
+		if ( this.onBeforeMatrixUpdate !== Object3D.prototype.onBeforeMatrixUpdate ) {
+
+			this.matrixData.updateMatrixWorldBefore = this.onBeforeMatrixUpdate.bind( this );
+
+		}
+
+		if ( this.onAfterMatrixUpdate !== Object3D.prototype.onAfterMatrixUpdate ) {
+
+			this.matrixData.updateMatrixWorldAfter = this.onAfterMatrixUpdate.bind( this );
+
+		}
+
+		if ( this.updateMatrixWorld !== Object3D.prototype.updateMatrixWorld ) {
```

Comment:
Please add the `// @deprecated` comment to this `if` clause.

Reply:
OK